src/compile.rs

```diff
@@ -945,7 +1017,13 @@ fn create_synthetic_resource(
     // Virtual resources become concrete when associated to concrete types
     dup_res_decl.is_virtual = dup_res_decl.is_virtual && dom_info.is_virtual;
     let dup_res_is_virtual = dup_res_decl.is_virtual;
-    dup_res_decl.annotations = Annotations::new();
+    // The synthetic resource keeps some, but not all annotations from its parent.
+    // Specifically, Makelist and derive are kept from the parent
+    dup_res_decl
+        .annotations
+        .annotations
+        .retain(|a| a.name.as_ref() == "makelist" || a.name.as_ref() == "derive");
```

Comment:
It would be less error-prone to use a dedicated method for `enum AnnotationInfo`. This way we must define this behavior when adding a new annotation.